pennylane/operation.py

```diff
@@ -355,16 +355,22 @@ def classproperty(func):
 
 
 def _process_data(op):
+    def _mod_and_round(x, mod_val):
+        if mod_val is None:
+            return x
+        return qml.math.round(qml.math.real(x) % mod_val, 10)
+
     # Use qml.math.real to take the real part. We may get complex inputs for
     # example when differentiating holomorphic functions with JAX: a complex
     # valued QNode (one that returns qml.state) requires complex typed inputs.
```

Comment:
@dwierichs this is out of scope for this PR, but it reminds me that we should add a comment here that the modulo done below is with respect to _expectation values_, not tapes that return the quantum state.

Reply:
Since this was added, this information is now stored under the `parameter_frequency` attribute.

Reply:
We could also switch to overriding `hash` for the relevant parametrized operators, instead of having to maintain a list like this here and check it for *all* operators.